brave-core/src/main/java/com/github/kristofa/brave/LocalTracer.java

```diff
@@ -62,9 +65,22 @@ static Builder builder(LocalTracer source) {
 
         abstract Builder spanCollector(SpanCollector spanCollector);
 
+        abstract Builder allowNestedLocalSpans(boolean allowNestedLocalSpans);
+
         abstract Builder traceSampler(Sampler sampler);
 
-        abstract LocalTracer build();
+        abstract LocalTracer autoBuild();
+
+        LocalTracer build() {
+            LocalTracer tracer = autoBuild();
+            if (tracer.allowNestedLocalSpans() && !(tracer.spanAndEndpoint().state() instanceof InheritableState)) {
+                throw new IllegalStateException(
+                        String.format("Cannot allowNestedLocalSpans for state %s that does not implement %s",
```

Comment:
this is over abstraction. Please remove this interface and use only the bridge class you initially defined. I don't want to dig a deeper hole by now making it possible to customize a hack!!